### ISSUE TITLE
Add Group Validations

### ISSUE THREAD

**Issue by esoterik** [YaleSTC/vesta#81]
- Draw that the group belongs to should match the leader and all members
- The size must match an available room size in the Draw (this might be complicated for situations where suites in multiple draws are all taken during an earlier draw)
- Number of members + leader must not exceed the size
- Open/full status should be accurate
- Groups can only be locked if they are full

~~Blocked on #49~~

**Comment by esoterik** [comment]
opening a new issue for handling the situation where a suite size becomes unavailable after group creation
